readthedocs/filetreediff/__init__.py

```diff
@@ -58,20 +175,58 @@ def get_diff(version_a: Version, version_b: Version) -> FileTreeDiff | None:
         manifests.append(manifest)
 
     # pylint: disable=unbalanced-tuple-unpacking
-    version_a_manifest, version_b_manifest = manifests
-    files_a = set(version_a_manifest.files.keys())
-    files_b = set(version_b_manifest.files.keys())
-
-    files_added = list(files_a - files_b)
-    files_deleted = list(files_b - files_a)
-    files_modified = []
-    for file_path in files_a & files_b:
-        file_a = version_a_manifest.files[file_path]
-        file_b = version_b_manifest.files[file_path]
+    current_version_manifest, base_version_manifest = manifests
+    current_version_file_paths = set(current_version_manifest.files.keys())
+    base_version_file_paths = set(base_version_manifest.files.keys())
+
+    current_version_build = Build.objects.get(id=current_version_manifest.build.id)
+    base_version_build = Build.objects.get(id=base_version_manifest.build.id)
+
+    files = []
+    resolver = Resolver()
+    for file_path in current_version_file_paths - base_version_file_paths:
+        files.append(
+            FileTreeDiffFile(
+                current_version=current_version,
+                current_version_build=current_version_build,
+                base_version=base_version,
+                base_version_build=base_version_build,
+                filename=file_path,
+                status=FileTreeDiffFileStatus.ADDED,
+                resolver=resolver,
+            )
+        )
```

Comment:
My suggestion mentioned before to simplify this:

```suggestion
        FileTreeDiff.append_file(
          filename=file_path,
          status=FileTreeDiffFileStatus.ADDED,
        )
```

Then, the `FileTreeDiff.append_file`:


```
def append_file(self, filename, status):
  self.files(FileTreeDiffFile(filename, status, diff=self, resolver=self.resolver)
```

This way we don't need to duplicate versions on each of the files and we are not exposing the internals to the caller. They just need to call `.append_file` without knowing the underlying details (base versions, resolver, etc)